src/apps/api/views/competitions.py

```diff
@@ -29,6 +29,6 @@ class PhaseViewSet(ReadOnlyModelViewSet):
     serializer_class = serializers.PhaseSerializer
 
 
-class SubmissionViewSet(ReadOnlyModelViewSet):
+class SubmissionViewSet(ModelViewSet):
```

Comment:
Can you create a submission for a competition you are not a participant of?

Reply:
Currently yes. I've considered this and it comes down to either assuming the phase we're given ( Which won't be from the user ), is for the current competition, or explicitly defining the competition and the phase, and double checking them against or each other or some such which seems unnecessary

Reply:
Yeah, phase is tied to competition so no need to specify competition.

I would add a check to see if user is in the competition 

```
if not phase.competition.participants.filter(user=request.user).exists():
    raise PermissionError("something")
```

Reply:
Alright, we'll keep this in mind